src/System.IO.Packaging/src/System/IO/Packaging/ZipWrappingStream.cs

```diff
@@ -88,25 +84,14 @@ public override void Flush()
             _baseStream.Flush();
         }
 
-        public override long Position
+        public override long Length
         {
             get
             {
-                return _baseStream.Position;
-            }
-            set
-            {
-                _baseStream.Position = value;
-            }
-        }
-
-        // For some strange reason, if the package FileAccess == Read,
-        // then can't get length from the stream.  Not supported.
-        // The workaround - write the stream to a memory stream, then return the length of the
-        // memory stream.
-        public override long Length
-        {
-            get {
+                // For some strange reason, if the package FileAccess == Read,
+                // then can't get length from the stream.  Not supported.
+                // The workaround - write the stream to a memory stream, then return the length of the
+                // memory stream.
                 if (_packageFileAccess == FileAccess.Read)
```

Comment:
Okay:

Whilst running the test project, the length property is called 177 times.
The special case code for `FileAccess == Read` is called 1 (!) time.

Confirmed: this code is necessary. If we remove it, then test `T139_Access_Length_Prop_Of_Stream` fails at line `729`

To me, it seems that this code is a cornercase, and isn't called very much